### ISSUE TITLE
Unreasonably slow operations

### ISSUE THREAD

**Issue by scott-griffiths** [scott-griffiths/bitstring#295]
This case was highlighted here: https://stackoverflow.com/questions/20845686/python-bit-array-performant in which bitstring is two orders of magnitude slower than some alternatives.

It's not meant to be as low-level as those alternatives, but now with bitarray being used under the hood it should be more competitive, but isn't yet.

It would be useful to profile this snippet to see what's going on.
```
  bs = bitstring.Bits(uint=0, length=64)
  for _ in range(100000):
      bs | bs
```

**Comment by scott-griffiths** [comment]
This is already partly addressed in the work for the 4.2 release. Worth profiling the difference again though.

**Comment by scott-griffiths** [comment]
This is considerably better in 4.2.3. It's still slow relatively, mainly due to the code only really profiling the object creation (it doesn't actually do anything useful). The same code in bitformat is even faster so I think that's a reasonable way to continue optimizations.
